### PR TITLE
retouch: various fixes and enhancements

### PR DESCRIPTION
**Pull request by elstoc**
added a new (local) keyboard shortcut to show / hide drawn shapes

resolved the following issues:

- clicking on the shape creation buttons while the module was disabled
  previously only toggled the shape button (user had to then enable
  the module to actually create a shape)

- show/edit shapes button was being activated inappropriately (when shapes
  were not being shown, when selecting earlier history items, when resetting
  module parameters and sometimes when adding a new shape)

- clicking on the show/edit shape button while in shape creation mode wasn't
  working - it now toggles shape creation modes off and switches to edit mode

### CODE REVIEW

**Comment by elstoc** [comment]
N.B. The one issue I could not resolve was that if you have the retouch shapes in edit mode, click on an earlier history item, and then go back to the top of the stack, the 'show and edit shapes' button is active but the shapes are not shown. You have to click twice to get it back to being consistent.

The only resolution I could find for this problem would be to reset the current gui module to NULL whenever selecting items on the history stack but I wasn't sure whether that might cause problems for other modules (perhaps crop & rotate?). This solution would ensure that the 'show and edit shapes' button is not active.

**Comment by TurboGit** [comment]
>     * clicking on the shape creation buttons while the module was disabled
>       previously only toggled the shape button (user had to then enable
>       the module to actually create a shape)

It was activated as soon a shape was actually drawn.

**Comment by TurboGit** [comment]
Ok, looks good to me.

File: src/iop/retouch.c [hunk 1]
```diff
@@ -3499,6 +3569,10 @@ void connect_key_accels(dt_iop_module_t *module)
 
   closure = g_cclosure_new(G_CALLBACK(_continuous_add_path_key_accel), (gpointer)module, NULL);
   dt_accel_connect_iop(module, "continuous add path", closure);
+
+  //show or hide
+  closure = g_cclosure_new(G_CALLBACK(_show_or_hide_accel), (gpointer)module, NULL);
+  dt_accel_connect_iop(module, "retouch show or hide", closure);
```
Comment (TurboGit) [inline src/iop/retouch.c:3575]: I would rename this to "retouch show or hide shapes"

**Comment by elstoc** [comment]
Agreed - I just blindly copied the one from spot removal without noticing it was a bit non-obvious what it did.

What do you think about removing gui focus when clicking the history stack? Too much just to fix a minor issue with retouch? I'm not sure there's another way to do it but it is a bit of an edge case.

**Comment by TurboGit** [comment]
I would let the code as-is for now. I'm wondering why the registered callback when the togglebutton is set is not called. I would have supposed that calling gtk_toggle_set_active() would have the same effect than clicking on the button?

File: src/iop/retouch.c (hunk 1 above)
Comment (TurboGit) [reply src/iop/retouch.c]: In fact better to have "show or hide shapes"

**Comment by TurboGit** [comment]
> I just blindly copied the one from spot removal without noticing it was a bit non-obvious

Right, "spot show or hide" is clear, we are speaking about spots. I have proposed another better choice.

**Comment by elstoc** [comment]
There's a lot of odd stuff in retouch but it felt like picking it apart was probably going to do more harm than good so I tried to be surgical. Happy to leave the remaining fault for now since it'll probably never happen in active use.

**Comment by TurboGit** [comment]
Looks better now. Thanks a lot for all your contributions.

**Comment by TurboGit** [comment]
And I have finally unified the spot removal accel to use the same wording.

**Comment by elstoc** [comment]
Might be worth a mention in the release notes - since that will have broken the key mapping for anyone using it already.

**Comment by elstoc** [comment]
Thanks @TurboGit I see you added a section to the release notes but you referenced crop&rotate instead of spot removal.

**Comment by TurboGit** [comment]
@elstoc : Indeed, fixed, thanks for catching that.

**Pull request merged**